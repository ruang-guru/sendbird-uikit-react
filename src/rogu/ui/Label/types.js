export const Typography = {
  H_1: 'H_1',
  H_2: 'H_2',
  H_3: 'H_3',
  SUBTITLE_1: 'SUBTITLE_1',
  SUBTITLE_2: 'SUBTITLE_2',
  BODY_1: 'BODY_1',
  BODY_2: 'BODY_2',
  BODY_3: 'BODY_3',
  BUTTON_1: 'BUTTON_1',
  BUTTON_2: 'BUTTON_2',
  CAPTION_1: 'CAPTION_1',
  CAPTION_2: 'CAPTION_2',
  CAPTION_3: 'CAPTION_3',
  TYPING_INDICATOR: 'TYPING_INDICATOR',
};

export const Colors = {
  ONBACKGROUND_1: 'ONBACKGROUND_1',
  ONBACKGROUND_2: 'ONBACKGROUND_2',
  ONBACKGROUND_3: 'ONBACKGROUND_3',
  ONBACKGROUND_4: 'ONBACKGROUND_4',
  ONBACKGROUND_5: 'ONBACKGROUND_5',
  ONCONTENT_1: 'ONCONTENT_1',
  ONCONTENT_2: 'ONCONTENT_2',
  PRIMARY: 'PRIMARY',
  ERROR: 'ERROR',
  SECONDARY_3: 'SECONDARY_3',
};
