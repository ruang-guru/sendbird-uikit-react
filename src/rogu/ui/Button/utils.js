import { Type, Size } from './type';

export function changeTypeToClassName(type) {
  switch (type) {
    case Type.PRIMARY: return 'rogu-button--primary';
    case Type.SECONDARY: return 'rogu-button--secondary';
    case Type.DANGER: return 'rogu-button--danger';
    case Type.DISABLED: return 'rogu-button--disabled';
    default: return null;
  }
}

export function changeSizeToClassName(size) {
  switch (size) {
    case Size.BIG: return 'rogu-button--big';
    case Size.SMALL: return 'rogu-button--small';
    default: return null;
  }
}
