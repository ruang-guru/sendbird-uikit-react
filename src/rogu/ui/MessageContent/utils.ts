const colorSet = {
  "#DF4141": ["A", "B", "C", "D"],
  "#61CE5E": ["E", "F", "G", "H"],
  "#6073E2": ["I", "J", "K", "L"],
  "#F89825": ["M", "N", "O", "P"],
  "#2EB5C0": ["Q", "R", "S", "T"],
  "#BB58D0": ["U", "V", "W", "X"],
  "#00A5FF": ["Y", "Z"],
};

export const generateColorFromString = (str: string): string => {
  const firstChar = str[0] || "";
  const normalizedFirstChar = firstChar.toUpperCase();

  let color = "inherit";
  for (const [hex, chars] of Object.entries(colorSet)) {
    if (chars.includes(normalizedFirstChar)) {
      color = hex;
      break;
    }
  }

  return color;
};

export default generateColorFromString;
