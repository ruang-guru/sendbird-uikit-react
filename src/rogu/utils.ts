export const isAssignmentMessage = (customType: string): boolean => customType === 'assignment';

export const isMaterialMessage = (customType: string): boolean => customType === 'material';

export const assignmentCtaToWeb = (cta:string):string=> {
    const listOfSerials = cta.split("&");
    const workspaceSerial = listOfSerials[1].split("=")[1];
    const classroomSerial = listOfSerials[2].split("=")[1];
    const assignmentSerial = listOfSerials[3].split("=")[1];
    const url = `https://kelas.ruangguru.com/workspace/${workspaceSerial}/classroom/${classroomSerial}/assignment/detail/${assignmentSerial}`;
    
    return url;
};


export const convertAssignmentDueUTCtoLocale = (dueAt:string):string => {
    
    const localeDate = new Date(dueAt).toLocaleString();
    console.log(dueAt, localeDate);

    return localeDate;
}
