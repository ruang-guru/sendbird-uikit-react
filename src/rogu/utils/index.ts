export * from "./assignmentMaterial";
export * from "./date";
export * from "./fileSize";
export * from "./fileType";
export * from "./message";
