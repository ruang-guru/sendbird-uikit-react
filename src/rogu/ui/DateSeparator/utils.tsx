import { differenceInCalendarDays, format, getDay } from "date-fns";

import { getDayString, DayStringType } from "../../utils";

export type DateSeparatorStringsType = {
  today: string;
  yesterday: string;
  days: DayStringType;
};

export const getDateSeparatorDifference = (
  createdAt: Date | number,
  strings: DateSeparatorStringsType
): string => {
  const diffWithToday = differenceInCalendarDays(new Date(), createdAt);

  if (diffWithToday === 0) {
    return strings.today;
  } else if (diffWithToday === 1) {
    return strings.yesterday;
  } else if (diffWithToday <= 7) {
    return getDayString(getDay(createdAt), strings.days);
  } else {
    return format(createdAt, "dd/MM/yyyy");
  }
};

export default getDateSeparatorDifference;
