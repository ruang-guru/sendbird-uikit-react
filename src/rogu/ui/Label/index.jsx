import React from 'react';
import PropTypes from 'prop-types';

import './index.scss';
import { Typography, Colors } from './types';
import { changeTypographyToClassName, changeColorToClassName } from './utils';
import getStringSet from './stringSet';

export default function Label({ children, className, color, style, type }) {
  return (
    <span
      className={[
        ...(Array.isArray(className) ? className : [className]),
        'sendbird-label',
        type ? changeTypographyToClassName(type) : '',
        color ? changeColorToClassName(color) : '',
      ].join(' ')}
      style={style}
    >
      {children}
    </span>
  );
}

Label.propTypes = {
  className: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.arrayOf(PropTypes.string),
  ]),
  type: PropTypes.oneOf([...Object.keys(Typography), '']),
  color: PropTypes.oneOf([...Object.keys(Colors), '']),
  children: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.number,
    PropTypes.element,
    PropTypes.any,
  ]),
  style: PropTypes.object,
};

Label.defaultProps = {
  className: [],
  type: '',
  color: '',
  children: null,
  style: undefined,
};

const LabelTypography = Typography;
const LabelColors = Colors;
const LabelStringSet = getStringSet('en');
export { LabelTypography, LabelColors, LabelStringSet };
