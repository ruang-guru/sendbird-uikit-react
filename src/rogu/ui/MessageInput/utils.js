// https://davidwalsh.name/javascript-debounce-function
// Returns a function, that, as long as it continues to be invoked, will not
// be triggered. The function will be called after it stops being called for
// N milliseconds. If `immediate` is passed, trigger the function on the

import { isUrl } from "../../../utils";

// leading edge, instead of the trailing.
export function debounce(func, wait, immediate) {
  let timeout;
  return function _debounce() {
    const context = this;
    // eslint-disable-next-line prefer-rest-params
    const args = arguments;
    const later = () => {
      timeout = null;
      if (!immediate) func.apply(context, args);
    };
    const callNow = immediate && !timeout;
    clearTimeout(timeout);
    timeout = setTimeout(later, wait);
    if (callNow) func.apply(context, args);
  };
};


export function getUrlFromWords(inputValue, setUrl){
  let inputValueArray = inputValue.split(/\s+/);
  let url = inputValueArray.find(word => isUrl(word));
  let hasUrl = !!url;
  return hasUrl && setUrl({hasUrl: true, text: url});
};



export default debounce;
