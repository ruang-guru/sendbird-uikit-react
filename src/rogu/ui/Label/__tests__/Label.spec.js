import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import Label, { LabelTypography, LabelColors } from '../index.jsx';
import { changeTypographyToClassName, changeColorToClassName } from '../utils.js';

describe('Label', () => {
  it('should have default classname if props are not provided', function () {
    const text = 'Example';
    const component = shallow(
      <Label
        type={LabelTypography.H_1}
        color={LabelColors.ONBACKGROUND_1}
      >
        {text}
      </Label>
    );
    expect(
      component.find(".sendbird-label").hasClass("sendbird-label--h-1")
    ).toBeTruthy();
    expect(
      component.find(".sendbird-label").hasClass("sendbird-label--color-onbackground-1")
    ).toBe(true);
    expect(
      component.find(".sendbird-label").text()
    ).toEqual(text)
  });

  it('should have each typography', function () {
    for (let color in LabelColors) {
      for (let typography in LabelTypography) {
        const component = shallow(<Label color={color} type={typography}>Component</Label>);
        expect(
          component.find('.sendbird-label').hasClass(changeColorToClassName(color))
        ).toBeTruthy();
        expect(
          component.find('.sendbird-label').hasClass(changeTypographyToClassName(typography))
        ).toBe(true);
      }
    }
  });

  it('should create a snapshot of a default label component', function () {
    const component = renderer.create(
      <Label
        type={LabelTypography.H_1}
        color={LabelColors.ONBACKGROUND_1}
      >
        Example
      </Label>,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
