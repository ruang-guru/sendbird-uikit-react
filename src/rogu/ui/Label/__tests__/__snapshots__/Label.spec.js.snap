// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Label should create a snapshot of a default label component 1`] = `
<span
  className="sendbird-label sendbird-label--h-1 sendbird-label--color-onbackground-1"
>
  Example
</span>
`;
